### ISSUE TITLE
`predict_coordinates` suggestions

### ISSUE THREAD

**Issue by gregcaporaso** [qiime2/q2-sample-classifier#15]
> would it be better to do this as a multilabel regression?
> currently each dimension is predicted separately

I think it would make sense to predict both together, rather than independently, since some variable could theoretically change independently across the two axes. Have you been able to find any literature on this problem? It's not something I've ever tried to do before, but it seems like there are probably some known best practices. 

> or precise location within 2-D physical space, such as the built environment

Not something you need to deal with now, but would 3-D physical space ultimately make more sense to support, with the second and third dimensions being optional? 

The `latitude` and `longitude` parameter names might make more sense as something like `axis1_category` and `axis2_category` (still with the same default values) if you're thinking that you'd like to use this for built environments, etc, where lat/long probably aren't the right axes. Or, just focus the docs for the moment on predicting lat/long. In either case, I think `_category` should be part of the parameter names for consistency with your other methods, and so it's more clear to users what these values are supposed to be.

**Comment by nbokulich** [comment]
>would it be better to do this as a multilabel regression?
currently each dimension is predicted separately

@benkaehler , could you offer some advice on this? I think multilabel/multioutput regression could theoretically increase accuracy if covariation between each target is taken into account. As far as I can tell from [the sklearn docs](http://scikit-learn.org/stable/modules/multiclass.html#multioutput-regression), `MultiOutputRegressor` does not do this, however, so we may be stuck.

**Comment by BenKaehler** [comment]
Hi Everyone, my understanding is that scikit-learn doesn't offer anything more sophisticated than providing multiple independent regressors for multioutput problems.

An alternative (that I didn't get time to look into for the feature-classifier) is to use [structured learning](http://pystruct.github.io/intro.html#intro). Apparently there are several possible libraries to use for structured learning, but I though the best-looking Python one was [pystruct](http://pystruct.github.io). It does have some licensing issues though.

Let me know if you would like me to look into it any further, or if you find a better solution!

**Comment by nbokulich** [comment]
> It does have some licensing issues though

From [pystruct's github page](https://github.com/pystruct/pystruct), it appears to be BSD-2. Would this be incompatible with BSD-3?

**Comment by BenKaehler** [comment]
It's not pystruct that's the problem, it's the libraries that it relies upon. See pystruct/pystruct#192 for details.

**Comment by nbokulich** [comment]
Have changed `latitude` and `longitude` to `axis1_category` and `axis2_category`.

Utilizing pystruct is still unresolved... will leave this issue open while I explore options for structured learning.

**Comment by gregcaporaso** [comment]
>  it appears to be BSD-2. Would this be incompatible with BSD-3?

No, it is compatible.

**Comment by nbokulich** [comment]
`predict_coordinates` is being removed from this plugin, so I am deleting this issue.
